api/handlers_test.go

```diff
@@ -136,6 +139,44 @@ func TestCreateSessionHandlerFunc(t *testing.T) {
 	})
 }
 
+func TestGetSessionHandlerFunc(t *testing.T) {
+	Convey("Given a request to retrieve a stored session", t, func() {
+		sessionID := "123"
+		currentTime := time.Now()
+		mockCache := &CacheMock{
+			GetFunc: func(email string) (*session.Session, error) {
+				return &session.Session{
+					ID:    "123",
+					Email: email,
+					Start: currentTime,
+				}, nil
+			},
+		}
+
+
+		getVars := func(r *http.Request)map[string]string {
+			return map[string]string{
+				"ID": sessionID,
+			}
+		}
+
+		sessionHandler := GetSessionHandlerFunc(mockCache, getVars)
+
+		req := httptest.NewRequest("GET", "/session/123", nil)
+		resp := httptest.NewRecorder()
+
+		Convey("When the request is handled by the router", func() {
```

Comment:
We arent testing the router here we are testing the handler - so something like _When the request is received_